test/timers/test-timers-reliability.js

```diff
@@ -1,54 +0,0 @@
-'use strict';
-// FaketimeFlags: --exclude-monotonic -f '2014-07-21 09:00:00'
-
-require('../common');
-
-const Timer = process.binding('timer_wrap').Timer;
-const assert = require('assert');
-
-let timerFired = false;
-let intervalFired = false;
-
-/*
- * This test case aims at making sure that timing utilities such
- * as setTimeout and setInterval are not vulnerable to time
- * drifting or inconsistent time changes (such as ntp time sync
- * in the past, etc.).
- *
- * It is run using faketime so that we change how
- * non-monotonic clocks perceive time movement. We freeze
- * non-monotonic time, and check if setTimeout and setInterval
- * work properly in that situation.
- *
- * We check this by setting a timer based on a monotonic clock
- * to fire after setTimeout's callback is supposed to be called.
- * This monotonic timer, by definition, is not subject to time drifting
- * and inconsistent time changes, so it can be considered as a solid
- * reference.
- *
- * When the monotonic timer fires, if the setTimeout's callback
- * hasn't been called yet, it means that setTimeout's underlying timer
- * is vulnerable to time drift or inconsistent time changes.
- */
```

Comment:
Is the monotonic-ness of the clock tested elsewhere? Or is the test completely unnecessary now?

Reply:
We never run this test and it has not worked since 8.x... imo it belongs in libuv which we consume.

Reply:
Got it, thanks.

Reply:
This test hasn't really been useful as it is pretty difficult to run it correctly.

Reply:
FWIW, it is definitely unfortunate that these tests do not work out of the box on all platforms. However I don't think testing the monotonic nature of the timers _only_ belongs in libuv.

For instance, introducing a regression by using `Date.now()` instead of `getLibuvTimeNow()` at https://github.com/nodejs/node/pull/20894/files/bb1dafb648f294edcc0c126de7f029f0239e6f36#diff-0a5d4868b2b9b17cf9e2c11f1bd1311eR183 would be caught by such a test.

It is unlikely, and I'm not saying we shouldn't get rid of this test, but coming up with a better way to run a similar test is probably a worthy (and interesting!) goal too.